### PR TITLE
Add micro-ROS RMW error handling API reference

### PR DESCRIPTION
**Pull request by pablogs9**
Signed-off-by: Pablo Garrido <pablogs9@gmail.com>

### CODE REVIEW

File: docs/rst/microros_documentation/user_api/user_api_middleware.rst [hunk 1]
```diff
@@ -210,3 +210,83 @@ A secondary API is provided to ping the Agent with a specific rmw configuration.
         // micro-ROS Agent is not available
         ...
     }
+
+Middleware error handling
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_middleware.rst:214]:
```suggestion
Error handling
```
Middleware is implicit on the tutorial title

File: docs/rst/microros_documentation/user_api/user_api_middleware.rst [hunk 2]
```diff
@@ -210,3 +210,83 @@ A secondary API is provided to ping the Agent with a specific rmw configuration.
...
+-------------------------
+
+micro-ROS RMW can be configured to report middleware error to user space using custom callbacks.
```
Comment (Acuadros95) [inline docs/rst/microros_documentation/user_api/user_api_middleware.rst:217]:
```suggestion
micro-ROS RMW can be configured to report middleware errors to user space using custom callbacks.
```

**Review by Acuadros95 — APPROVED** [review]

File: docs/rst/microros_documentation/user_api/user_api_middleware.rst [hunk 3]
```diff
@@ -210,3 +210,83 @@ A secondary API is provided to ping the Agent with a specific rmw configuration.
         // micro-ROS Agent is not available
         ...
     }
+
+Error handling
+-------------------------
+
+micro-ROS RMW can be configured to report middleware errors to user space using custom callbacks.
```
Comment (dcoello-eProsima) [inline docs/rst/microros_documentation/user_api/user_api_middleware.rst:217]: what happens if this option is disabled and a middleware error occurs?

Comment (pablogs9) [reply docs/rst/microros_documentation/user_api/user_api_middleware.rst:217]: 8dc261ea584a6750e9f534c5e6a175f8a26913b4

**Review by dcoello-eProsima — APPROVED** [review]

**Pull request merged**